### PR TITLE
Fix jhlang and react rtl support

### LINKED ISSUE #7786: Navbar menu Item Not translate  to selected lang
##### **Overview of the issue**
when change lang to another from menu the menu item not set to selected lang and translated value

##### **JHipster Version(s)**

```
samplebeta-02@0.0.0 /Users/developer/Desktop/samplebeta02
└── (empty)

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "5.0.0-beta.2",
    "applicationType": "monolith",
    "baseName": "samplebeta02",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "memcached",
    "websocket": "spring-websocket",
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "searchEngine": "elasticsearch",
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "replaced-by-jhipster-info",
    "clientFramework": "react",
    "useSass": true,
    "clientPackageManager": "yarn",
    "testFrameworks": [
      "protractor"
    ],
    "jhiPrefix": "jhi",
    "otherModules": [],
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "fa"
    ]
  }
}
</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

##### **Environment and Tools**

java version "1.8.0_60"
Java(TM) SE Runtime Environment (build 1.8.0_60-b27)
Java HotSpot(TM) 64-Bit Server VM (build 25.60-b23, mixed mode)

git version 2.16.2

node: v8.11.2

npm: 5.6.0

yeoman: 2.0.2

yarn: 1.7.0

Docker version 17.05.0-ce, build 89658be

docker-compose version 1.13.0, build 1719ceb

##### **Browsers and Operating System**

- [ ] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by agaspardcilia**
fix #7786 
For some reason, in the languages sub-gen, `this.config.set('languages', _.union(this.currentLanguages, this.languagesToApply));` was not updating languages during the generation but after. It was preventing the subgen to properly add the new desired language.

On the other side, RTL support was a bit messed up for React. I fixed it because it was preventing the needle from being printed in the file when there was no RTL languages selected.

- Please make sure the below checklist is followed for Pull Requests.

- [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [x] Tests are added where necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by agaspardcilia** [comment]
I'm aware of #7815 so I tried to not messed things up for you @vishal423 and I hope I was able to. :sweat_smile:

**Comment by vishal423** [comment]
@agaspardcilia , I see few conflicting changes as I removed the `require.context` use and loading merged bundles with axios get call (bundles are merge during webpack build like we do in angular).

**Comment by agaspardcilia** [comment]
Alright, I'll take a look.

**Comment by agaspardcilia** [comment]
@vishal423 I'll wait for your PR to be merged and then I'll do what's necessary to use your stuff. Just don't mind this PR.

**Comment by deepu105** [comment]
https://github.com/jhipster/generator-jhipster/pull/7815 is merged @agaspardcilia

File: generators/generator-base-private.js [hunk 1]
```diff
@@ -211,13 +211,14 @@ module.exports = class extends Generator {
      * @param languages
      */
     updateLanguagesInLanguagePipe(languages) {
-        if (this.clientFramework !== 'angularX') {
+        if (!['angularX', 'react'].includes(this.clientFramework)) {
```
Comment (deepu105) [inline generators/generator-base-private.js]: I think this check can be removed as we only have angularX and react now

**Comment by jdubois** [comment]
Is this OK? If yes, let's merge this

**Comment by agaspardcilia** [comment]
@jdubois afaik, yes it is.

**Pull request merged**